### PR TITLE
Adds Emax Targets

### PR DESCRIPTION
**Pull request by EmaxModel**
Hello, I am an engineer from Emax. We are engaged in the development of FPV and Radio. Recently, our team is committed to the research of expresslrs and has designed several different hardware. We want to make this product more perfect, so we want to participate in the research of this product with you. I hope we can join, thank you!

### CODE REVIEW

**Comment by JyeSmith** [comment]
Hi @EmaxModel 

Can you please have a team member contact me on discord and we can arrange a group discussion :smile:

**Comment by EmaxModel** [comment]
uh, our code can't be merged. Do you want us to send samples to you

**Comment by JyeSmith** [comment]
Before merging we would like to test samples and confirm all features operate as expected.  It will also allow us to test future developments, write documents, and perform user support.

File: src/include/target/EMAX_2400_RX.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#define GPIO_PIN_BUSY           5
+#define GPIO_PIN_DIO0           -1
+#define GPIO_PIN_DIO1           4
+#define GPIO_PIN_MOSI           13
+#define GPIO_PIN_MISO           12
+#define GPIO_PIN_SCK            14
+#define GPIO_PIN_RST            2
+#define GPIO_PIN_LED            16
+#define GPIO_PIN_BUTTON         0
+#define timerOffset             -1
```
Comment (pkendall64) [inline src/include/target/EMAX_2400_RX.h:17]: This line is not used

File: src/targets/emax_2400.ini [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+src_filter = ${env_common_esp82xx.src_filter} -<tx_*.cpp>
+
+[env:EMAX_2400_RX_via_BetaflightPassthrough]
+extends = env:EMAX_2400_RX_via_UART
+upload_protocol = custom
+upload_speed = 420000
+upload_command = ${env_common_esp82xx.bf_upload_command}
+
+[env:EMAX_2400_RX_via_WIFI]
+extends = env:EMAX_2400_RX_via_UART
```
Comment (pkendall64) [inline src/targets/emax_2400.ini:59]: Not required as it should be a configurator alias as said above

File: src/lib/SCREEN/OLED/oledscreen.cpp [hunk 3]
```diff
@@ -218,18 +218,46 @@ void OLEDScreen::updateIdleScreen(uint8_t dirtyFlags)
...
+            u8g2.drawStr(70,45, &(ratio_string[current_ratio_index])[0]);
+            u8g2.drawStr(0,60, power.c_str());
+            u8g2.setFont(u8g2_font_profont10_mr);
+            u8g2.drawStr(70,56, "TLM");
+            u8g2.drawStr(0,27, "Ver: ");
+            char buffer[7];
+            strncpy(buffer, version, 6);
+            buffer[6] = 0;
+            u8g2.drawStr(38,27, buffer);
+        #endif
```
Comment (pkendall64) [inline src/lib/SCREEN/OLED/oledscreen.cpp:260]: I don't think we'd be happy with making target specific changes here. We're trying to get away from them and work to feature toggles instead. Also, the regulatory domain is displayed in the LUA script. If we do more work of OLED/TFT screens for non-opentx based radios and it becomes necessary to display the regulatory domain then we'd do it for all OLED/TFT capable devices.

File: src/include/target/EMAX_2400_RX.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#ifndef DEVICE_NAME
+#define DEVICE_NAME "EMAX 2400RX"
+#endif
+
```
Comment (JyeSmith) [inline src/include/target/EMAX_2400_RX.h:4]: Has the inductor been included in the sx1280 circuit?

**Comment by CapnBry** [comment]
* Did you folks seriously use GPIO2 for the S.PORT connection instead of GPIO13 to save a resistor? GPIO2 is a strapping pin and should not be used. Connecting to a wide range of handsets we're almost certainly going to run into some that pull the SPORT pin in a way you're not expecting which then could create issues accessing the bootloader. I'm not sure why you would make this decision to put your hardware in such a perilous compatibility position simply to save one of the cheapest components on the board.
* All the `TARGET_TX_EMAX_900` should be removed, as there is no "special code" for them.
* The receiver targets are simply clones and should be removed from the targets list and added only to the configurator as clones.
* The 900MHz modules you're saying is rated for 250mW but can output 2000mW? That's a pretty big difference from what you're rating it for (MaxPower) vs what you're saying it can do (HighPower). I feel like these two values need to be closer unless you don't have much confidence in your design? Selling the user a 250mW TX with a wink that says it can go to 2W (and possibly burn itself out) is a pretty big cop out on the quality of your own design.
* The Nano and regular TX modules have all the same pin mappings so they should probably use the same header file?

My biggest concern by far is the use of GPIO2 which may cause compatibility issues and therefore a disproportionate amount of support issues and quality concerns for the project.

**Comment by EmaxModel** [comment]
> * Did you folks seriously use GPIO2 for the S.PORT connection instead of GPIO13 to save a resistor? GPIO2 is a strapping pin and should not be used. Connecting to a wide range of handsets we're almost certainly going to run into some that pull the SPORT pin in a way you're not expecting which then could create issues accessing the bootloader. I'm not sure why you would make this decision to put your hardware in such a perilous compatibility position simply to save one of the cheapest components on the board.
> * All the `TARGET_TX_EMAX_900` should be removed, as there is no "special code" for them.
> * The receiver targets are simply clones and should be removed from the targets list and added only to the configurator as clones.
> * The 900MHz modules you're saying is rated for 250mW but can output 2000mW? That's a pretty big difference from what you're rating it for (MaxPower) vs what you're saying it can do (HighPower). I feel like these two values need to be closer unless you don't have much confidence in your design? Selling the user a 250mW TX with a wink that says it can go to 2W (and possibly burn itself out) is a pretty big cop out on the quality of your own design.
> * The Nano and regular TX modules have all the same pin mappings so they should probably use the same header file?
> 
> My biggest concern by far is the use of GPIO2 which may cause compatibility issues and therefore a disproportionate amount of support issues and quality concerns for the project. EDIT: This doesn't match the schematic provided so is this PR a draft then?

1. Crsf pin I don't know that using GPIO2 will lead to some unexpected problems. Since such problems will occur, we will modify it.
2. I wish you were talking about target_ RX_ EMAX_ 900 instead of target_ TX_ EMAX_ 900, because in target_ TX_ EMAX_ 900, power_ OUTPUT_ Values these values are set according to our own circuit board design. I will merge nano and regular files.
3. The target header file of the receiver has been deleted and will be added in the configuration file.
4. Both nano and regular modules will be merged into the same header file.
Don't worry, the GPIO2 pin will be modified, and we will provide the best hardware design.
thank you!

**Pull request closed without merging**